flags/src/main/java/com/yahoo/vespa/flags/custom/ClusterCapacity.java

```diff
@@ -0,0 +1,89 @@
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * @author freva
+ */
+// @Immutable
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(value = JsonInclude.Include.NON_NULL)
+public class ClusterCapacity {
+    private final int count;
+    private final double vcpu;
+    private final double memoryGb;
+    private final double diskGb;
+    private final Optional<Double> bandwidthGbps;
+
+    @JsonCreator
+    public ClusterCapacity(@JsonProperty("count") int count,
+                           @JsonProperty("vcpu") double vcpu,
+                           @JsonProperty("memoryGb") double memoryGb,
+                           @JsonProperty("diskGb") double diskGb,
+                           @JsonProperty("bandwidthGbps") Double bandwidthGbps) {
+        this.count = (int) requireNonNegative("count", count);
+        this.vcpu = requireNonNegative("vcpu", vcpu);
+        this.memoryGb = requireNonNegative("memoryGb", memoryGb);
+        this.diskGb = requireNonNegative("diskGb", diskGb);
+        this.bandwidthGbps = Optional.ofNullable(bandwidthGbps);
+    }
+
+    /** Returns a new ClusterCapacity equal to {@code this}, but with the given count. */
+    public ClusterCapacity withCount(int count) {
+        return new ClusterCapacity(count, vcpu, memoryGb, diskGb, bandwidthGbps.orElse(null));
+    }
+
+    @JsonGetter("count") public int count() { return count; }
+    @JsonGetter("vcpu") public double vcpu() { return vcpu; }
+    @JsonGetter("memoryGb") public double memoryGb() { return memoryGb; }
+    @JsonGetter("diskGb") public double diskGb() { return diskGb; }
+    @JsonGetter("bandwidthGbps") public Double bandwidthGbpsOrNull() { return bandwidthGbps.orElse(null); }
+
+    @JsonIgnore
+    public double bandwidthGbps() { return bandwidthGbps.orElse(1.0); }
+
+    @Override
+    public String toString() {
+        return "ClusterCapacity{" +
+                "count=" + count +
+                ", vcpu=" + vcpu +
+                ", memoryGb=" + memoryGb +
+                ", diskGb=" + diskGb +
+                ", bandwidthGbps=" + bandwidthGbps +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        ClusterCapacity that = (ClusterCapacity) o;
+        return count == that.count &&
+                Double.compare(that.vcpu, vcpu) == 0 &&
+                Double.compare(that.memoryGb, memoryGb) == 0 &&
+                Double.compare(that.diskGb, diskGb) == 0 &&
+                ((bandwidthGbps.isEmpty() && that.bandwidthGbps.isEmpty()) ||
+                        ((bandwidthGbps.isPresent() && that.bandwidthGbps.isPresent() &&
+                                Double.compare(that.bandwidthGbps.get(), bandwidthGbps.get()) == 0)));
```

Comment:
I decided to use Double.compare because it was used to test for equality by the auto-generated code for all the double fields.  According to Effective Java, Double.compare should be used to compare doubles to 1. take care of +/- 0.0 and NaN and 2. to avoid auto-boxing.  With Optional<Double> (2) is not relevant.
I switched to using OptionalDouble which uses Double.compare in equals.